test/source/mock/lib/api.ts

```diff
@@ -144,19 +146,24 @@ export class Api<REQ, RES> {
     if (!req.url) {
       throw new Error('no url');
     }
-    if (this.handlers[req.url]) {
+    const attesterHandler = getMockAttesterEndpoints(this.attesterConfig);
+    const allHandlers: Handlers<REQ, RES> = {
+      ...(attesterHandler as Handlers<REQ, RES>),
```

Comment:
Well, these `REQ` and `RES` keep popping up because this `Api` class was meant to be generic, but now this generic class is supposed to know about certain "attester"s. That's too specific and contradicts the initial design of this class.
So it'd be better for this class to query some other injected function or interface for additional handlers. This function or object can be injected in the constructor.

Reply:
If I understand correctly, you mean something like this?
Or you want something else?

```ts
interface AdditionalHandlerProvider<REQ, RES> {
  getHandlers(): Handlers<REQ, RES>;
}

public constructor(apiName: string, protected handlers: Handlers<REQ, RES>, protected urlPrefix = '', protected additionalHandlerProvider?: AdditionalHandlerProvider<REQ, RES>) {
}

// In chooseHandler funciton
  const additionalHandlers = this.additionalHandlerProvider?.getHandlers() || {};
  const allHandlers: Handlers<REQ, RES> = {
    ...additionalHandlers,
    ...this.handlers,
  };


class AttesterHandlerProvider<REQ, RES> implements AdditionalHandlerProvider<REQ, RES> {
  constructor(private attesterConfig: AttesterConfig) {}

  getHandlers(): Handlers<REQ, RES> {
    return getMockAttesterEndpoints(this.attesterConfig) as Handlers<REQ, RES>;
  }
}

```

Reply:
Hmm... we can't pass it in the constructor, because the initialization is tucked too deep in `startAllApisMock`

It'd be better just to have one property e.g. `public configProvider: ConfigurationProviderInterface<REQ,  RES>` instead of `handlers`. This interface is what you described as `interface AdditionalHandlerProvider`

The class `ConfigurationProvider` (what you suggested as `AttesterHandlerProvider`) doesn't need to be generic, it should only support actual REQ and RES our handlers use, that is `{ query: { [k: string]: string }; body?: unknown }` and `unknown`, see https://github.com/FlowCrypt/flowcrypt-browser/pull/4952/files

We can initialize `ConfigurationProvider` with `AttesterConfig` etc. and wire it to the concrete instance of Api with `t.mockApi.configProvider = new ConfigurationProvider { attesterConfig: ... }` instead of current `t.mockApi.attesterConfig =`
Their types will match because the concrete `mockApi` types match our handlers and hence the ConfigurationProvider type too

Reply:
In my opinion, I don't think this approach will be effective. If we only set up `mockApi.configProvider` during `Api` initialization, the `/attester/pub` and `/attester/ldap` endpoints will not receive live `attesterConfig` data.

For example, if a developer sets `t.mockApi.attesterConfig = x` at the beginning of a test, and then updates it to `t.mockApi.attesterConfig = y` during the test, the current configuration will not be reflected when the test calls `/attester/pub` or `/ldap`. Instead, these endpoints will only retrieve the previous value.

What do you think?

Reply:
I meant to not expose `t.mockApi.attesterConfig` any more.
Instead, 
`t.mockApi.configProvider = new ConfigProvider({ attesterConfig: x, ...}}`

`t.mockApi.configProvider = new ConfigProvider({ attesterConfig: y, ...}}`

or 
`t.mockApi.configProvider.attesterConfig = y;`

Reply:
Yep!. You are right. We don't need to expose `attesterConfig` anymore.

Reply:
I propose to do this in next PR as this PR is not small so that it will be easier for you to review.